crates/sui-source-validation-service/src/lib.rs

```diff
@@ -311,9 +314,31 @@ pub struct ErrorResponse {
 }
 
 async fn api_route(
+    headers: HeaderMap,
     State(app_state): State<Arc<AppState>>,
     Query(Request { address, module }): Query<Request>,
 ) -> impl IntoResponse {
+    let version = headers
+        .get(SUI_SOURCE_VALIDATION_VERSION_HEADER)
+        .as_ref()
+        .and_then(|h| h.to_str().ok())
+        .map(|s| s.to_string());
+
+    match version {
```

Comment:
Do we return the version we used in the response header? That might be helpful for the"no version set" case.

Additionally, in the versioning scheme for RPC, we only allow people to specify the major/minor version (not the patch) to give us some flexibility to release things under a patch version and not break all clients.  Is that worth porting over to this case?  (This was another reason why we supplied the actual version used in the response header).

Reply:
> Do we return the version we used in the response header?

Added! Discovered that headers are [normalized to lowercase](https://docs.rs/http/latest/http/header/index.html) without much control inside the framework. I think this is done because of [spec](https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2)? 

> only allow people to specify the major/minor version 

Done!